sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -40,16 +39,38 @@ private[execution] sealed trait HashedRelation {
   /**
     * Returns matched rows.
     */
-  def get(key: InternalRow): Seq[InternalRow]
+  def get(key: InternalRow): Iterator[InternalRow]
 
   /**
     * Returns matched rows for a key that has only one column with LongType.
     */
-  def get(key: Long): Seq[InternalRow] = {
+  def get(key: Long): Iterator[InternalRow] = {
     throw new UnsupportedOperationException
   }
 
   /**
+   * Returns the matched single row.
+   */
+  def getValue(key: InternalRow): InternalRow
+
+  /**
+   * Returns the matched single row with key that have only one column of LongType.
+   */
+  def getValue(key: Long): InternalRow = {
+    throw new UnsupportedOperationException
+  }
+
+  /**
+   * Returns true iff all the keys are unique.
+   */
+  def allUnique: Boolean
+
+  /**
+   * Returns a read-only copy of this, to be safely used in current thread.
+   */
+  def copy(): HashedRelation
```

Comment:
when i saw this being called, my first guess was that a deep copy was performed. we should rename it to make it seem lighter weight. maybe asReadOnlyCopy? similar to the one on ByteBuffer.